app/src/main/java/communi/dog/aplicatiion/AddMarkerActivity.java

```diff
@@ -105,8 +105,8 @@ private String getMarkerTitle(boolean isDogsitter, boolean isFood, boolean isMed
         return msg;
     }
 
-    private void backToMap() {
-        Intent backToMapIntent = new Intent(this, MapScreenActivity.class);
+    private void backButton() {
+        Intent backToMapIntent = new Intent(this, ProfilePageActivity.class);
```

Comment:
what about the cases when entering this screen from the map??
you need to find a way to distinguish between cases we get there from the map and from the profile
for example you can add Boolean extra to the calling intent when getting from profile and check its value

Reply:
Added a new commit that should address this issue.